spec/models/organisation_spec.rb

```diff
@@ -1,74 +1,79 @@
 describe Organisation do
-  context 'associations' do
-    it { is_expected.to have_many(:users) }
-    it { is_expected.to have_many(:locations) }
+  it { is_expected.to have_many(:users) }
+  it { is_expected.to have_many(:locations) }
 
-    context 'when deleting an organisation as an admin' do
-      let(:org) { create(:organisation) }
-      let!(:user) { create(:user, organisation: org) }
-      let!(:location) { create(:location, organisation: org) }
-      let!(:ip) { Ip.create(address: "1.1.1.1", location: location) }
+  context 'when deleting an organisation' do
+    let(:org) { create(:organisation) }
+    let!(:user) { create(:user, organisation: org) }
+    let!(:location) { create(:location, organisation: org) }
+    let!(:ip) { Ip.create(address: "1.1.1.1", location: location) }
 
-      it 'removes all of its children ' do
-        org.destroy
-        expect { user.reload }.to raise_error ActiveRecord::RecordNotFound
-        expect { location.reload }.to raise_error ActiveRecord::RecordNotFound
-        expect { ip.reload }.to raise_error ActiveRecord::RecordNotFound
-      end
+    before { org.destroy }
+
+    it 'removes all associated users' do
+      expect { user.reload }.to raise_error ActiveRecord::RecordNotFound
+    end
+
+    it 'removes all associated locations' do
+      expect { location.reload }.to raise_error ActiveRecord::RecordNotFound
     end
 
-    context 'When an organisation signs up under the same name as another organisation' do
-      before { create(:organisation, name: 'Gov Org 1') }
+    it 'removes all associated ip addresses' do
+      expect { ip.reload }.to raise_error ActiveRecord::RecordNotFound
+    end
+  end
+
+  context 'when signing up an organisation with the same name as another registered organisation' do
+    before { create(:organisation, name: 'Gov Org 1') }
 
-      it 'is not valid' do
-        organisation = build(:organisation, name: 'Gov Org 1')
-        expect(organisation).not_to be_valid
-      end
+    it 'is not valid' do
+      organisation = build(:organisation, name: 'Gov Org 1')
+      expect(organisation).not_to be_valid
     end
+  end
 
-    context 'When updating the organisations service email' do
-      it 'does not validate the organisations name' do
-        organisation = create(:organisation)
-        organisation.service_email = 'newserviceemail@gov.uk'
-        expect(organisation).to be_valid
-      end
+  context 'when updating an organisation service email' do
+    it 'remains valid' do
```

Comment:
What does this actually test?

It creates an organisation, then updates the service email to a valid email and checks its valid? Seems unnecessary.

The original spec is `it 'does not validate the organisations name' do`. Which implies that an existing organisation with an invalid name, when updating the service email, does not validate the name again.